objects/src/accounts/storage/mod.rs

```diff
@@ -242,110 +281,77 @@ impl AccountStorage {
             return Err(AccountError::StorageSlotIsReserved(index));
         }
 
-        // only value slots of basic arity can currently be updated
-        match self.layout[index as usize] {
-            StorageSlotType::Value { value_arity } => {
-                if value_arity > 0 {
-                    return Err(AccountError::StorageSlotInvalidValueArity {
-                        slot: index,
-                        expected: 0,
-                        actual: value_arity,
-                    });
-                }
-            },
-            slot_type => Err(AccountError::StorageSlotNotValueSlot(index, slot_type))?,
-        }
```

Comment:
This is what I'm calling a bugfix. I don't understand why there was such a constraint in place to begin with. tagging @bobbinth

Reply:
The main reason for this was to keep `AccountStorage::set_item()` parallel to the `account::set_item` procedure in the transaction kernel - so, I'm not sure this is a bug. But looking at this now it does look like we have 2 issues:

1. There are no similar methods for updating storage maps (e.g., `set_map_item()`).
2. It seems like in `AccountStorage::apply_delta()` we are ignoring all `StorageMap`-related deltas. That's definitely a bug and I'm a bit surprised we don't have a test for it.

cc @Dominik1999

Reply:
But we don't test applying the delta in this PR either, right? Or am I missing something?

I can push a test for it to this PR @hackaugusto

Reply:
I would probably do a separate PR to fix `AccountStorage ::apply_delta()` and add a test for it.

Reply:
its merged into this PR now